CONTRIBUTING.md

```diff
@@ -1,54 +1,50 @@
-- Please only start coding after we have agreed to a specification to avoid putting lots of work into something which may not get accepted later.
+- To avoid putting lots of work into something which may not get accepted later, please start coding after we have agreed to a specification.
 
 ## Jamulus project/source code general principles
 
-### Stability
+### 1. Stability
 
-This has been, and must continue to be, the most important requirement. It is particularly important for use cases such as the Worldjam where instability during a live performance would be unacceptable. The following two principles are designed to support this principle.
+Stability has been, and must continue to be, the most important requirement. In particular, instabilities during live performances such as WorldJam are not acceptable. The following principles are designed to support this.
 
-### [Keep it Simple and Stupid](https://en.wikipedia.org/wiki/KISS_principle) and [Do One Thing and Do It Well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well)
+### 2. [Keep it Simple and Stupid](https://en.wikipedia.org/wiki/KISS_principle) and 3. [Do One Thing and Do It Well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well)
 
-If a feature or function can be achieved in another way by another system or method, then that is preferable to building that feature into Jamulus.
+If a feature or function can be achieved in another way by another system or method, then that is preferable instead of building that feature into Jamulus. Consider having a look at e.g. the [JSON-RPC](https://github.com/jamulussoftware/jamulus/blob/master/docs/JSON-RPC.md) API which allows you to communicate with the client and server from the outside.
 
 ### Source code consistency
 
-- Respect the existing code style: Tab size=4, insert spaces.
-- Space before and after `(` and `)`, except no space between `)` and `;`, and no space before an empty `()`.
-- All bodies of `if`, `else`, `while`, `for`, etc., to be enclosed in braces `{` and `}`, on separate lines.
+Our CI will fail and tell you about styling violations.
 
-You can use your editor's or IDE's clang-format support to accomplish that.
-On the command line, you can run `make clang_format` to do the same before committing.
+To keep a consistent coding style, please install `clang-format` on your PC and run it before committing.
 
-Do not use diff/patch to send your code changes but create a Github fork of the Jamulus code and create a Pull Request when you are done.
+There are multiple ways to run clang-format:
 
-Please run a local build test. Make sure there are no errors. After opening a pull request, keep an eye on the CI checks for quality or compile issues and fix them as required.
+- Use your editor's or IDE's clang-format support to accomplish that.
+- Via make: after running `qmake Jamulus.pro`, run `make clang_format`
+- Manually: run `clang-format -i <path/to/changed/files>`
 
```

Comment:
I think we should specify where to find the clang-format version which we use in CI (maybe not hardcode the version here though).
#2559 + related for context.

Reply:
Probably yes? But a link to the CI (+ comment) would maybe help for less maintenance burden...

Reply:
Added a suggestion.